rispy/utils.py

```diff
@@ -0,0 +1,37 @@
+"""Miscellaneous functions."""
+
+from typing import Dict, List
+from copy import deepcopy
+
+from .config import TYPE_OF_REFERENCE_MAPPING
+
+
+def invert_dictionary(mapping: Dict) -> Dict:
+    """Invert the keys and values of a dictionary."""
+    remap = {v: k for k, v in mapping.items()}
+    if len(remap) != len(mapping):
+        raise ValueError("Dictionary cannot be inverted; some values were not unique")
+    return remap
+
+
+def pretty_reference_types(
+    reference_list: List,
+    reverse: bool = False,
+    strict: bool = False,
+    type_map: Dict = TYPE_OF_REFERENCE_MAPPING,
+) -> List:
+    """Convert RIS reference types to pretty names."""
+
+    def convert(ref, d=type_map):
+        old_type = ref["type_of_reference"]
+        try:
+            ref["type_of_reference"] = d[old_type]
+        except KeyError:
+            if strict and old_type not in d.values():
+                raise KeyError(f'Type "{old_type}" not found.')
+        return ref
+
+    if not reverse:
+        return [convert(r) for r in deepcopy(reference_list)]
```

Comment:
seems odd that this mutates the reference. Maybe it'd be better if it just returned the reference type as a string, and a user could modify the reference object if they want to?

Reply:
It does seem odd but it indeed mutates the reference. If a user wanted to return the reference type as a string, they could just use `map[ref["type_of_reference"]]`, I made this specifically for modifying them.

Reply:
If you want to keep as is, you could update the docs and docstring to indicate that this returns a copy of references with the key modified, I didnt expect this behavior and only saw it when I read the code

Reply:
I'll also expose a function that just returns the string, since it's already in there and someone may find it useful